hbw/app/javascript/packs/hbw/components/shared/context/store.jsx

```diff
@@ -22,14 +22,67 @@ const withStoreContext = (WrappedComponent) => {
       this.initStore();
     }
 
+    getTaskById = async (data) => {
+      const { env } = this.props;
+      const incomingTask = await env.connection.request({
+        url:  `${env.connection.serverURL}/tasks/${data.task_id}`,
+        data: {
+          cache_key:    data.cache_key,
+          entity_class: env.entity_class,
+        },
+      });
+
+      const tasks = this.state.tasks.filter(task => task.id !== incomingTask.id);
+      tasks.push(incomingTask);
+      this.setState({
+        tasks:    (this.orderTasks(tasks)),
+        fetching: false,
+        ready:    true
+      });
+
+      return incomingTask;
+    };
+
+    onCreated = (data) => {
+      this.getTaskById(data);
+    };
+
+    onAssigned = async (data) => {
+      const { translator: t } = this.props.env;
+      const task = await this.getTaskById(data);
+
+      Application.messenger.notice(t('notifications.new_assigned_task', {
+        task_name: task.name
+      }));
+    };
+
+    onComplete = (data) => {
+      const tasks = this.state.tasks.filter(task => task.id !== data.task_id);
+      this.setState({ tasks });
+    };
+
+    onReceived = (data) => {
+      if (data.event_name === 'create') {
+        this.onCreated(data);
+      }
+
+      if (data.event_name === 'assignment') {
+        this.onAssigned(data);
+      }
+
+      if (['complete', 'delete'].includes(data.event_name)) {
+        this.onComplete(data);
```

Comment:
I think it's not cool to pass faceless `data` to all functions. You have structure (event_name, task_id and cache_key) so pass explicitly data you really needed in function, please.